src/networkProvider/transactionStatus.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * An abstraction for handling and interpreting the "status" field of a transaction.
+ */
+export class TransactionStatus {
+    /**
+     * The raw status, as fetched from the Network.
+     */
+    readonly status: string;
+
+    /**
+     * Creates a new TransactionStatus object.
+     */
+    constructor(status: string) {
+        this.status = (status || "").toLowerCase();
+    }
+
+    /**
+     * Creates an unknown status.
+     */
+    static createUnknown(): TransactionStatus {
+        return new TransactionStatus("unknown");
+    }
+
+    /**
+     * Returns whether the transaction is pending (e.g. in mempool).
+     */
+    isPending(): boolean {
+        return (
+            this.status == "received" ||
+            this.status == "pending" ||
+            this.status == "partially-executed"
```

Comment:
can we have status `partially-executed`? I cannot find it here:
https://github.com/ElrondNetwork/elrond-go-core/blob/main/data/transaction/status.go
and I can only find it in a unit test on elrond-proxy-go